server/lib/genome/importers/api_importers/civic/api_client.rb

```diff
@@ -4,12 +4,32 @@
 module Genome; module Importers; module ApiImporters; module Civic
   class ApiClient
 
-    def enumerate_variants
-      variant_edges = send_query
+    def enumerate_drugs
+      drug_edges = send_query(DrugsQuery).therapies.edges
       Enumerator.new do |y|
-        until variant_edges.empty?
-          variant_edges.each { |edge| y << edge }
-          variant_edges = send_query(variant_edges[-1].cursor)
+        until drug_edges.empty?
```

Comment:
This works just fine as written, so no need to change it unless you want to. But just FYI, you can also request 

```
    pageInfo {
      hasNextPage
      endCursor
    }
```

At the top level of any of these queries. This avoids bringing the cursor back for each item, and just gives you the cursor of the last one in the page, and then provides a boolean `hasNextPage`. That way you can set up a loop sort of like (pseudocode) `while hasNextPage == true; send_query(after: endCursor)`.  Which will save you having to send a final query which comes back empty.

The overhead of the extra cursors and the one additional request is minimal so I don't think its worth rewriting, just something to know for next time.

Reply:
@acoffman am I misunderstanding how this works? this query gives me `"hasNextPage": false` in the playground even though I assume there are > 27 drugs

```
query {
  therapies(first: 26, after: "") {
    pageInfo {
      endCursor
      hasNextPage
    }
    edges {
      node {
        id
        name
        ncitId
      }
    }
  }
}
```

Reply:
@jsstevenson You do understand it correctly, turns out there is actually a small issue with the SQL powering this specific query. I have a fix ready and it should go out today or tomorrow. Sorry for the inconvenience!

Reply:
(what a relief, figured it was either that or i was going insane)